internal/postgres/convert_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"harbourbridge/spanner/ddl"
+	"strings"
+	"testing"
+
+	pg_query "github.com/lfittl/pg_query_go"
+	nodes "github.com/lfittl/pg_query_go/nodes"
+	"github.com/stretchr/testify/assert"
+)
+
+// This file contains very basic tests of Cvt API functionality.
+// Most of the Cvt APIs are also tested in process_test.go (where
+// they are tested using data from schema/data conversion).
+
+func TestSetSchemaMode(t *testing.T) {
+	cvt := MakeCvt()
+	cvt.SetSchemaMode()
+	assert.True(t, cvt.schemaMode())
+	assert.False(t, cvt.dataMode())
+}
+
+func TestSetDataMode(t *testing.T) {
+	cvt := MakeCvt()
+	cvt.SetDataMode()
+	assert.False(t, cvt.schemaMode())
+	assert.True(t, cvt.dataMode())
+}
+
+func TestGetDDL(t *testing.T) {
+	cvt := MakeCvt()
+	cvt.sSchema["table1"] = ddl.CreateTable{
+		"table1",
+		[]string{"a", "b"},
+		map[string]ddl.ColumnDef{
+			"a": ddl.ColumnDef{Name: "a", T: ddl.Int64{}},
+			"b": ddl.ColumnDef{Name: "b", T: ddl.Float64{}},
+		},
+		[]ddl.IndexKey{ddl.IndexKey{Col: "a"}},
+		"",
+	}
+	cvt.sSchema["table2"] = ddl.CreateTable{
+		"table2",
+		[]string{"a"},
+		map[string]ddl.ColumnDef{
+			"a": ddl.ColumnDef{Name: "a", T: ddl.Int64{}},
+		},
+		[]ddl.IndexKey{ddl.IndexKey{Col: "a"}},
+		"",
+	}
+	ddl := cvt.GetDDL(ddl.Config{})
+	assert.Equal(t, 2, len(ddl))
+	normalize := func(s string) string {
+		return strings.Join(strings.Fields(s), " ")
+	}
+	if len(ddl) >= 2 {
```

Comment:
Why are you checking this when you have ``assert.Equal(t, 2, len(ddl))`` three lines earlier?

Reply:
This code has gone away: see "Fix flaky TestGetDLL (convert_test)" 

But to answer your question: assert doesn't stop execution. Instead it reports and error and keeps on going. Hence, if is needed to avoid a crash.